### PR TITLE
Author View Backend Endpoints

### LINKED ISSUE #18: Feature SALES-2: Implement markAll(author) API
Implement an API endpoint that allows for all of the sales relating to an author to be marked as paid.

### LINKED ISSUE #21: Feature AUTHOR-1: Implement Author endpoints (3.2)
Implement endpoints:

3.2
   -  /: searchAuthor() → AuthorSearchResponse
   -  /: getByAuthor() → PagedSaleResponse

And aggregator endpoint for total unpaid amount for an author.

### PR DESCRIPTION
**Pull request by michaelscutari**
Implementation of backend author-related endpoints, including author search and mark all paid. Design decisions discussed below, as always I am happy to receive feedback.

Endpoints added:
- `GET /api/sales/author-payments` --> Returns `AuthorPaymentGroupResponse`
- `PUT /api/sales/author-payments/mark-paid`
- `GET /api/books/authors` --> Returns `PagedResponse<String>`

DTOs added:
- `AuthorPaymentGroupResponse` --> Group of author payments, with calculated total unpaid royalties. 
- `AuthorPaymentSaleResponse` --> Nested within `AuthorPaymentGroupResponse`
- `MarkAllPaidRequest`
- `MarkAllPaidResponse`

Originally, I was considering creating a new `/api/authors` grouping, however since authors are not a first-class data type within our current structure, I decided that keeping things to `/api/books` and `/api/sales` made more sense for now.

As for the split, I put the author search functionality into `/api/books` because author is the sole responsibility of `Books` and is unrelated to `Sales`. In the future, if we decide to add an `Authors` database table, I think that we could look at refactoring this.

Tests are basic and cover some common edge cases, including white-space removal as discussed in the requirements document.

Additional change to the `api::generate` handle in `package.json` to verify that the backend is running. Also now gives more explicit failure and completion messages.

Resolves #21 
Resolves #18

### CODE REVIEW

File: backend/src/main/java/edu/duke/bookpublishing/sales/dto/AuthorPaymentSaleResponse.java
Comment (Danny1021) [inline backend/src/main/java/edu/duke/bookpublishing/sales/dto/AuthorPaymentSaleResponse.java:1]: I see a lot of repeat between this and the generic sale response, the only difference being the inclusion of the book title and the author being included. Should we consider a merger of the two? Perhaps the sale table should also show the book title and author and we can just have the one DTO.

File: backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkAllPaidRequest.java
Comment (Danny1021) [inline backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkAllPaidRequest.java:1]: The example may not be accurate. The requirements outline the authors are a comma separated list in the case of multiple authors, so `Fitzegarld, F. Scott` would be interpreted as two authors despite that being one person. I imagine it would instead be "F. Scott Fitzgerald", unless the intent here was to have two authors (which would behave the same as one author per the req).

File: backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkPaidRequest.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package edu.duke.bookpublishing.sales.dto;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotBlank;
+
+/** Request body for marking all unpaid sales for an author as paid. */
+@Schema(description = "Request to mark all unpaid sales for an author as paid")
+public record MarkPaidRequest(
```
Comment (Danny1021) [inline backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkPaidRequest.java]: Maybe `MarkAllPaidRequest` would be a more fitting name? This could be confused with marking a single sale as paid to the author while this DTO is intended for marking all unpaid sales of an author as paid.

File: backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkPaidResponse.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package edu.duke.bookpublishing.sales.dto;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+
+/** Response body for marking all unpaid sales for an author as paid. */
+@Schema(description = "Response after marking an author's unpaid sales as paid")
+public record MarkPaidResponse(
```
Comment (Danny1021) [inline backend/src/main/java/edu/duke/bookpublishing/sales/dto/MarkPaidResponse.java]: Same here

File: backend/src/main/java/edu/duke/bookpublishing/sales/SaleService.java [hunk 3]
```diff
@@ -59,6 +64,55 @@ public Page<Sale> getPagedSales(LocalDate startDate, LocalDate endDate, Pageable
...
+        saleRows.add(AuthorPaymentSaleResponse.from(sale));
+        if (!Boolean.TRUE.equals(sale.getHasAuthorBeenPaid())) {
+          unpaidTotal = unpaidTotal.add(sale.getAuthorRoyalty());
+        }
+      }
+
+      groups.add(new AuthorPaymentGroupResponse(entry.getKey(), unpaidTotal, saleRows));
+    }
+
+    return groups;
```
Comment (Danny1021) [inline backend/src/main/java/edu/duke/bookpublishing/sales/SaleService.java:113]: Love this implementation!

**Review by Danny1021 — COMMENTED** [review]
Please see comments
